### PR TITLE
Distgen postfix separate

### PR DESCRIPTION
**Pull request by phracek**
This Pull Request removes all postfix stuff.
- Dockefile is rendered
- help.md is rendered
- README.md is rendered.

Build is working.

### CODE REVIEW

**Comment by TomasTomecek** [comment]
(I won't have time to review this PR)

**Comment by phracek** [comment]
No problem. In the meantime I will play with it a bit more. Including tests.
@mhlavink Any idea how to easily test dovecot?

**Comment by mhlavink** [comment]
Install dovecot, set mail_location in config, start it and try to 
connect to it with some imap client.

https://wiki.dovecot.org/TestInstallation

Dne 5.10.2017 v 12:52 Petr Hracek napsal(a):

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+sudo: required
+services:
+  - docker
+script:
+  - hooks/pre_build
+  # Docker Hub hack
+  - sudo cp -av ./Dockerfile.template ./Dockerfile
```
Comment (bkabrda) [inline .travis.yml]: I don't understand why the steps on lines 5-8 are required here. The rendering and build should be done correctly by the `make` call on line 8.

File: hooks/pre_build_centos [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash -ex
```
Comment (bkabrda) [inline hooks/pre_build_centos]: What is this file for?

**Review by bkabrda — CHANGES_REQUESTED** [review]

File: hooks/pre_build_centos (hunk 2 above)
Comment (phracek) [reply hooks/pre_build_centos]: It is used for CentOS, but this is future.

File: .travis.yml (hunk 1 above)
Comment (phracek) [reply .travis.yml]: hooks/pre_build move it to Dockerfile.template https://github.com/container-images/container-image-template/blob/master/hooks/pre_build#L1

Comment (phracek) [reply .travis.yml]: See https://github.com/container-images/tools/blob/master/.travis.yml#L5

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/TPgJ2](https://da.gd/TPgJ2)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/dWCZ8](https://da.gd/dWCZ8)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/zPBdk](https://da.gd/zPBdk)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/KBnc](https://da.gd/KBnc)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/LA2i](https://da.gd/LA2i)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/TSe4c](https://da.gd/TSe4c)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/ACDt](https://da.gd/ACDt)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/tK7R](https://da.gd/tK7R)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/ggM5M](https://da.gd/ggM5M)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/LKTeo](https://da.gd/LKTeo)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/w4Wc](https://da.gd/w4Wc)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/PgCFI](https://da.gd/PgCFI)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/JtGSR](https://da.gd/JtGSR)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/H4sB](https://da.gd/H4sB)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/piG1J](https://da.gd/piG1J)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/cjYPH](https://da.gd/cjYPH)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/1pVor](https://da.gd/1pVor)__
* Comment: __This is testing CI__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/7vMv](https://da.gd/7vMv)__
* Comment: __This is testing CI__

File: Makefile [hunk 3]
```diff
@@ -1,17 +1,46 @@
...
+	docker run -e MYHOSTNAME=localhost $(IMAGE_REPOSITORY)
+
+test: build
+	cd tests; MODULE=docker URL="docker=${IMAGE_REPOSITORY}" DOCKERFILE="../$(DOCKERFILE)" VERSION=${VERSION} DISTRO=${DISTRO} mtf -l *.py
+
+test-in-container: test-image
+	docker run --rm -ti -v /var/run/docker.sock:/var/run/docker.sock:Z -v ${PWD}:/src ${TEST_IMAGE_NAME} "SELECTORS=${SELECTORS}"
+
+test-image:
+	docker build --tag=${TEST_IMAGE_NAME} -f ./Dockerfile.tests .
```
Comment (bkabrda) [inline Makefile]: It seems that there's no `Dockerfile.tests` in this repo, so this target seems to be useless. I'd suggest removing it.

Comment (bkabrda) [reply Makefile]: + also the `test-in-container` target.

File: Makefile [hunk 4]
```diff
@@ -1,17 +1,46 @@
...
+dg:
+	${DG_EXEC} --template $(DOCKERFILE_SRC) --output $(DOCKERFILE)
+	${DG_EXEC} --template help/help.md --output help/help.md.rendered
+
+doc: dg
+	mkdir -p ./root/
+	${GOMD2MAN} -in=help/help.md.rendered -out=./root/help.1
+
+build: doc dg
+	sudo chown -R $(shell id -u) .
```
Comment (bkabrda) [inline Makefile]: I'm wondering if it'd make more sense to to create the rendered/generated files by using redirection, e.g. `command > Dockerfile` - that way we'd have resulting files owned by current user and would avoid the `sudo` call - does that make sense?

Comment (phracek) [reply Makefile]: I did not know, it is possible

**Review by bkabrda — APPROVED** [review]

**Comment by bkabrda** [comment]
LGTM now, merging. Nice work!

**Pull request merged**